iceoryx_hoofs/source/posix_wrapper/file_lock.cpp

```diff
@@ -92,8 +92,9 @@ FileLock& FileLock::operator=(FileLock&& rhs) noexcept
 
         CreationPattern_t::operator=(std::move(rhs));
 
-        m_name = std::move(rhs.m_name);
-        m_fd = std::move(rhs.m_fd);
+        // for details see here iceoryx_hoofs/include/iceoryx_hoofs/design_pattern/creation.hpp
+        m_name = std::move(rhs.m_name); // NOLINT(hicpp-invalid-access-moved)
+        m_fd = std::move(rhs.m_fd);     // NOLINT(hicpp-invalid-access-moved, hicpp-move-const-arg)
```

Comment:
In the line above ` CreationPattern_t::operator=(std::move(rhs));` is called therefore the linter thinks that `rhs` is no longer valid which is actually only partially true since only the creation pattern members are moved.

Reply:
The file does not contain an explanation for the lint but only for the move of the base class.

Wouldn't the proper solution be to just move the base class?
```cpp
CreationPattern_t::operator=(std::move(*static_cast<CreationPattern_t*>(&rhs)));
```
Yes, it's ugly but without the pointer detour the copy elision optimization is prevented ... and we really need to get rid of the creation pattern ;)